web/src/compose_validate.js

```diff
@@ -691,3 +691,24 @@ export function validate(scheduling_message) {
     }
     return validate_stream_message(scheduling_message);
 }
+
+export function warn_if_mentioning_private_recipient(mention_text, full_name, user_id) {
+    if (compose_state.get_message_type() !== "private") {
+        return mention_text;
+    }
+
+    const recipient = compose_state.private_message_recipient();
```

Comment:
I think this would be better calling `compose_pm_pill.get_user_ids`; this module already calls that function in another place, and then you wouldn't need to convert from emails.